### PR TITLE
#161967123 Implement JWT token authentication 

### PR DESCRIPTION
**Pull request by davishooly**
### What does this PR do?
  - Add JWT-authentication to Authors Haven Django App
<hr>

### Description of Task to be completed?
  - Add token to user registration response
  - Add token to user login response
 
<hr>

### How should this be manually tested?
 - Clone the repo and cd into the project folder
 - Create a virtual environment by running `$ python3 -m venv <name_of_environment>`
 - Activate the virtual environment by running `source <name_of_environment>/bin/activate`
 - Run `$ pip install -r requirements.txt`
 - Then add the following environment variables to your computer.

    -`$ export DB_NAME='choose_a_dbname'`
       -`$ export DB_USER='db_username'`
       -`$ export DB_PASSWORD='db_role_password'`
       -`$ export DB_HOST='localhost or preffered host'`
       -`$ export DB_PORT='server_port_number'`
       -`$ export SECET_KEY='secrete_key'`

- Change the current branch to ft-jwt-authentication-161967123
- Run `$ python manage.py runserver`
- Open postman then enter this route `http://localhost:8000/api/users` to register a user
- You will receive a response with a token 
<hr>

### What are the relevant pivotal tracker stories?
[ #161967123 ](https://www.pivotaltracker.com/story/show/161967123)
<hr>

#### Checklist
- My code follows the style guidelines of this project
- I have performed a self-review of my own code
- I have commented my code, particularly in hard-to-understand areas
- I have made corresponding changes to the documentation
- My changes generate no new warnings

#### Screenshot
screen shot 2018-12-06 at 09 53 42
screen shot 2018-12-05 at 18 15 08

### CODE REVIEW

File: authors/apps/authentication/tests/test_register.py [hunk 1]
```diff
@@ -91,18 +91,11 @@ def test_empty_password(self):
 
     def test_register_user(self):
         """ test register user """
-
         response = self.register_user(self.new_user)
-
         self.assertEqual(
             response.status_code,
             status.HTTP_201_CREATED
         )
```
Comment (bettblake08) [inline authors/apps/authentication/tests/test_register.py:98]: Add a test for the access_token variable in the response body.

File: authors/apps/authentication/tests/test_login.py [hunk 2]
```diff
@@ -61,7 +61,6 @@ def test_login_with_empty_password_field(self):
 
     def test_login(self):
         """ test login success """
-
         response = self.login(self.user)
```
Comment (bettblake08) [inline authors/apps/authentication/tests/test_login.py:64]: Add a test for the access_token in the response body.

**Review by bettblake08 — CHANGES_REQUESTED** [review]

File: authors/apps/authentication/backends.py [hunk 3]
```diff
@@ -1,9 +1,30 @@
-# import jwt
+import jwt
 #
```
Comment (Darius-Ndubi) [inline authors/apps/authentication/backends.py]: Style this imports by removing the unused ilines having '#'

File: authors/apps/authentication/tests/test_register.py (hunk 1 above)
Comment (davishooly) [reply authors/apps/authentication/tests/test_register.py:98]: okay am working on that

**Review by bettblake08 — APPROVED** [review]
Looks good to me!

**Comment by mutoromike** [comment]
Kindly attach screenshots from Postman.

**Review by Darius-Ndubi — CHANGES_REQUESTED** [review]
On the PR you have left us hanging. Your PR implements JWT but running python manage.py runserver does not test the functionality being added. Add more steps to how one can test the feature manually. Add screenshots too.

**Review by jillo-abdullahi — APPROVED** [review]
Please add options to do migrations before running the server and let testers know that they need to try to create a new user from postman to verify if this works. Otherwise, the actual JWT implementation looks good to me.

**Comment by davishooly** [comment]
> Kindly attach screenshots from Postman.

File: authors/apps/authentication/backends.py (hunk 3 above)
Comment (davishooly) [reply authors/apps/authentication/backends.py]: cool @Darius-Ndubi

**Review by Darius-Ndubi — APPROVED** [review]

**Comment by davishooly** [comment]
@mutoromike i have done the changes requested

**Review by mutoromike — CHANGES_REQUESTED** [review]
Attach a screenshot of the JSON data (i.e. Postman) that's posted when creating a user or when logging in.

**Review by denomash — APPROVED** [review]
works fine

**Comment by davishooly** [comment]
@mutoromike  cool mike

**Comment by davishooly** [comment]
@mutoromike I have added the screenshot for the JSON data

**Review by mutoromike — APPROVED** [review]

**Pull request merged**